src/qforte/abc/algorithm.py

```diff
@@ -258,6 +260,18 @@ def fill_pool(self):
         else:
             raise ValueError('Invalid operator pool type specified.')
 
+        # If possible, impose symmetry restriction to operator pool
+        try:
+            temp_sq_pool = qf.SQOpPool()
+            for sq_operator in self._pool_obj.terms():
+                create = sq_operator[1].terms()[0][1]
+                annihilate = sq_operator[1].terms()[0][2]
+                if symmetry_check(self._sys.orb_irreps_to_int, create, annihilate) == self._irrep:
+                    temp_sq_pool.add(sq_operator[0], sq_operator[1])
+            self._pool_obj = temp_sq_pool
+        except:
```

Comment:
It is exactly what you said. At the moment, symmetry is supported only for `system_type = molecule`. If someone uses `system_type = model` there will be no `orb_irreps_to_int` attribute in `self._sys`. Can you please clarify what you mean by "specifying exactly which kind of exception"? Should I just put a comment in the code explaining why this is exception is needed or have a print statement alerting the user that symmetry is not supported for `system_type = model`?

Reply:
No, I'm referring to [a feature of the Python language](https://docs.python.org/3/tutorial/errors.html#handling-exceptions). line 272 should be `except AttributeError:` so that only that kind of error is caught. You should also have a comment in the code to specify why an `AttributeError` means no symmetry.